### ISSUE TITLE
Multiple `#[test]` attributes run the test multiple times.

### ISSUE THREAD

**Issue by Ethiraric** [rust-lang/rust#90979]
If a function has multiple `#[test]` attributes, the test is run as many times as the attribute is repeated with `cargo test`.

The following code runs `doing_the_thing` 8 times:

```rust
mod tests {
    #[test]
    #[test]
    #[test]
    #[test]
    #[test]
    #[test]
    #[test]
    #[test]
    fn doing_the_thing() {}
}

fn main() {}
```

I expected to see this happen: The compiler errors

Instead, this happened: It compiles fine and the test is run 8 times

### Meta
`rustc --version --verbose`:
```
rustc 1.56.1 (59eed8a2a 2021-11-01)
binary: rustc
commit-hash: 59eed8a2aac0230a8b53e89d4e99d55912ba6b35
commit-date: 2021-11-01
host: x86_64-unknown-linux-gnu
release: 1.56.1
LLVM version: 13.0.0
```

Happens on nightly as well:
```
rustc 1.58.0-nightly (d914f17ca 2021-11-16)
binary: rustc
commit-hash: d914f17ca71a33a89b2dc3436fca51b1a091559e
commit-date: 2021-11-16
host: x86_64-unknown-linux-gnu
release: 1.58.0-nightly
LLVM version: 13.0.0
```

cc @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
I think we should emit a warning in such case, that shouldn't be too complicated.

@Ethiraric said he would be interested into working on it so I'll assign them and help them implementing it.

**Comment by chaitanyav** [comment]
@GuillaumeGomez can i work on this one?.

**Comment by GuillaumeGomez** [comment]
No, as indicated in my message, @Ethiraric is already working on it.

**Comment by ehuss** [comment]
FYI, PR #88681 is open to check for duplicate attributes.  It doesn't handle test/bench because those are proc-macros, and it can be tricky to detect duplicates on those (see https://github.com/rust-lang/rust/pull/88681#discussion_r748167322).  So just a heads up this may not be particularly easy.  I'm curious why it doesn't fail [this check](https://github.com/rust-lang/rust/blob/6414e0b5b308d3ae27da83c6a25098cc8aadc1a9/compiler/rustc_builtin_macros/src/test.rs#L110-L117) during subsequent passes.
